src/otaclient/ota_core.py

```diff
@@ -536,40 +542,70 @@ def _execute_update(self):
 
         # ------ pre-update ------ #
         logger.info("enter local OTA update...")
+
+        with TemporaryDirectory() as _tmp_dir:
+            use_inplace_mode = can_use_in_place_mode(
+                dev=self._boot_controller.standby_slot_dev,
+                mnt_point=_tmp_dir,
+                threshold_in_bytes=int(
+                    self._ota_metadata.total_regulars_size
+                    * STANDBY_SLOT_USED_SIZE_THRESHOLD
+                ),
+            )
+        logger.info(
+            f"check if we can use in-place mode to update standby slot: {use_inplace_mode}"
+        )
+
         self._boot_controller.pre_update(
             self.update_version,
-            standby_as_ref=False,  # NOTE: this option is deprecated and not used by bootcontroller
-            erase_standby=True,  # NOTE: as of now, we only have rebuild mode, so always erase_standby
+            # NOTE: this option is deprecated and not used by bootcontroller
+            # NOTE(20250602): will use this arg again for in-place mode in the future.
+            # TODO:(20250604): when standby_as_ref is set, skip mounting active slot.
+            standby_as_ref=use_inplace_mode,
+            erase_standby=not use_inplace_mode,
         )
-
         # prepare the tmp storage on standby slot after boot_controller.pre_update finished
-        self._resource_dir_on_standby.mkdir(exist_ok=True)
+        self._resource_dir_on_standby.mkdir(exist_ok=True, parents=True)
+        self._ota_tmp_meta_on_standby.mkdir(exist_ok=True, parents=True)
 
+        # NOTE(20250529): first save it to /.ota-meta, and then save it to the actual
+        #                 destination folder.
         logger.info("save the OTA image file_table to standby slot ...")
```

Comment:
Due to `inplace` mode will cleanup the standby slot during delta calculation, we cannot directly save the file_table to `/opt/ota/image-meta`, so we first save to a protected folder `/.ota-meta`, and later in post OTA update phase, we then move it into the `/opt/ota/image-meta`.